app/models/course/condition/achievement.rb

```diff
@@ -3,4 +3,40 @@ class Course::Condition::Achievement < ActiveRecord::Base
   belongs_to :achievement, class_name: Course::Achievement.name
 
   default_scope { includes(:achievement) }
+
+  validate :valid_achievement_condition, if: :new_record?
```

Comment:
Subtle bug that @allenwq spotted. On update, this record's achievement will already be among its conditional's achievements, so validations will always fail if they are enabled.

Now that you point it out, we do need validations on update, but they have to be slightly different... The list of conditions shouldn't include the old achievement.